mllib/src/main/scala/org/apache/spark/ml/feature/RFormulaParser.scala

```diff
@@ -269,40 +310,78 @@ private[ml] object RFormulaParser extends RegexParsers {
   }
 
   private val intercept: Parser[Term] =
-    "([01])".r ^^ { case a => Intercept(a == "1") }
+    "([01])".r.n ^^ { case a => Intercept(a == "1") }
```

Comment:
Pardon, what does `.n` do?

Reply:
Just introduced a shorter way of escaping whitespace around the tokens. Uses the implicit class defined inside the object on line 290:

```scala
  private val space = "[ \\n]*".r

  /* Utility function for skipping whitespace around a regex. */
  private implicit class RegexParserUtils(val r: Regex) {
    def n: Parser[String] = (space ~> r <~ space)
  }

```
Given a regexp, returns a string parser which skips the surrounding white space

Reply:
Oh I see it now. Hm, I think that will be a little more confusing than it's worth to readers. Just add space to the regexes for clarity, I'd say.

Reply:
Got it, I replaced it with a normal function call.